tests/python/gaia-ui-tests/gaiatest/apps/system/app.py

```diff
@@ -23,25 +26,38 @@ class System(Base):
     _software_home_button_locator = (By.ID, 'software-home-button')
 
     def wait_for_status_bar_displayed(self):
-        self.wait_for_element_displayed(*self._status_bar_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._status_bar_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
 
     def wait_for_notification_toaster_displayed(self, timeout=10, message=None):
-        self.wait_for_element_displayed(*self._notification_toaster_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._notification_toaster_locator))
+        Wait(self.marionette, timeout=timeout).until(
+            expected.element_displayed(element), message=message)
 
     def wait_for_notification_toaster_not_displayed(self, timeout=10):
-        self.wait_for_element_not_displayed(*self._notification_toaster_locator)
+        Wait(self.marionette, timeout=timeout, ignored_exceptions=StaleElementException).until(
+            lambda m: not m.find_element(*self._notification_toaster_locator).is_displayed())
```

Comment:
I had to use the ignored_exceptions=StaleElementException thing, because it would otherwise time out on that spot.
I'll try again, once I addressed all your review comments.